### ISSUE TITLE
display feedback on kanban

### ISSUE THREAD

**Issue by Drew-Macgibbon** [Drew-Macgibbon/nuxt-feedback#3]
info coming soon

**Comment by JapneetRajput** [comment]
- When the issue is being displayed as a card on the column : Title, Category, Assignee, Priority, Votes (Display current number of votes and if they click on it they will vote or unvote)

- When a single issue is clicked upon then more details can be shown : Title, Body, Category, Status, Assignee, Votes, github issue, status, category, Size.

- Kanban column is included in status issue : #5

**Comment by Drew-Macgibbon** [comment]
@JapneetRajput yes to all of it except assignee which might require more thought. 

It's worth remembering that we're not trying to recreate GH issues, just taking the parts that we need for a good feedback system.

I'm not sure showing the assignee will add value to our users, and if it's just for our reference we have GH for that.

Feel free to change my mind!

**Comment by JapneetRajput** [comment]
@Drew-Macgibbon yeah makes sense. 
We can skip the assignee field.

**Comment by JapneetRajput** [comment]
I've thought of two flows for displaying the feedbacks on the kanban board : 

1. We fetch all tasks (including all fields) and display them column(status) wise. If a user clicks on a card the card details modal will be displayed. As we already have the task details we can display them directly. 

Pros : 
- Lesser API calls.

Cons : 
- All the data that will be fetch may or may not be necessary.

2. We fetch all tasks (only those fields which will be displayed on card) and display them column(status) wise. If a user clicks on a card, we will hit our DB to fetch all the details of that particular task and display them. 

Pros : 
- We will only be fetching data that is necessary.

Cons : 
- More API calls.

@Drew-Macgibbon Which one shall we go with? or any other flow that you can think of
I'm also actively thinking of any more pros/cons for them and tricks to optimize them.

**Comment by Drew-Macgibbon** [comment]
@JapneetRajput 2nd option. We will have additional info such as comments fetched for the Modal, so additional calls are to be expected.

**Comment by aayu5hgit** [comment]
@Drew-Macgibbon @JapneetRajput 
I'm attaching a ref. image of the kanban board after the changes suggested by @Drew-Macgibbon

![image](https://github.com/incubrain/nuxt-feedback/assets/86314754/4d8ed484-79de-4f52-b2c8-b51943d9fd7c)

Will be enhancing this too, but after we fetch the dynamic data from `supabase`.

**Comment by aayu5hgit** [comment]
@Drew-Macgibbon @JapneetRajput I'll start working on (and probably finish too) the `modal` that is popped after the card is clicked.

I'll be playing with the **dummy data** I have created (for now); and when things will work smooth, we can integrate with the `supabase` data easily thereafter.

**Comment by aayu5hgit** [comment]
@Drew-Macgibbon @JapneetRajput 
I've started with the `modal` functionality and sharing the progress (attaching a ref. image)
Do let me know if things are to be modified / any other suggestions

![image](https://github.com/incubrain/nuxt-feedback/assets/86314754/c04675a2-b500-472a-b082-b7198dda3eed)

**Comment by Drew-Macgibbon** [comment]
@aayu5hgit nice progress. Nuxt/ui is so good to work with!

A few small UI suggestions, some of which you're probably already planning.

- center the Modal
- move category, priority, avatar etc. to the `UCard`  `#header` and the description to `#body` slots. In the future we will add comments to the `#footer`, I think this is a logical flow.
- we should aim to reduce the screen realestate used in the header section. Remove `due date`, that will be for internal use. Try to fit the rest in 2 rows, mobile first. `created at` can be `age` and use relative time like 5 days.

**Comment by aayu5hgit** [comment]
@Drew-Macgibbon I've made the suggested changes

- [x] center the Modal
- [x] move category, priority, avatar etc. to the `UCard`  `#header` and the description to `#body` slots. In the future we will add comments to the `#footer`, I think this is a logical flow.
- [x] we should aim to reduce the screen realestate used in the header section. Remove `due date`, that will be for internal use. Try to fit the rest in 2 rows, mobile first. `created at` can be `age` and use relative time like 2mo/2d/2h/2m.

![image](https://github.com/incubrain/nuxt-feedback/assets/86314754/13728cd6-deea-4c3d-85ec-ba92eaf89c69)

**Comment by Drew-Macgibbon** [comment]
@aayu5hgit nice, that will do for now. We will make continued improvements to the ui as we go.

I didn't realize the modal was already centered lol, the kanban threw me off.
